docs/dev-environment-setup.md

```diff
@@ -0,0 +1,239 @@
+
+`psql --list` to verify the databases were created.
+
+Rather than interacting with Sqitch directly, we use a [custom script](https://github.com/bcgov/cas-ciip-portal/blob/develop/schema/data/deploy-data.sh) for convenience to deploy the database, including seed data for dev, test and production.
+
+
+From the project root, run `.bin/deploy-data.sh --help` to see possible options. In development, this is most commonly run with `.bin/deploy-data.sh -d -dev` to drop the current database and redeploy after pulling down database-changing work from the remote or tampering with the database.
+
+## 7. Install and Run the Web Application
+
+Install the `node_modules` including `devDependencies` from the `package.json`:
+```
+cd app && yarn --dev
+```
+
+Run the development server with `yarn dev` and navigate in your browser to `localhost:3004`.
+
+Work that changes Relay GraphQL queries will require you to re-run `yarn build:relay` (which alternatively can be run using a `--watch` flag).
+
+### GraphQL Query Debugger
+
+Graphiql is an interactive in-browser GraphQL IDE available in development at `localhost:3004/graphiql`. It is handy for developing queries interactively before inserting them into a Relay fragment.
+
+### Voyager
+
+Voyager is an API graph explorer for GraphQL and is available in development at `localhost:3004/voyager`.
+
+## 7.5 Bypass Authentication in Development
+
+There are some shortcuts to bypass authentication in development and run the server while logged in as a preset user:
+
+```
+cd app && yarn dev <AS_USER>
+```
+...where `<AS_USER>` is one of the following:
+
+```
+AS_REPORTER
+AS_CERTIFIER
+AS_ANALYST
+AS_ADMIN
+```
+
+[Learn more here](https://github.com/bcgov/cas-ciip-portal/blob/develop/docs/all-views-inventory.md) about the functionality available to each user type.
+
+## 8. Run Tests
+
+### Unit Tests with Jest
+
+```
+cd app && yarn test
+```
+Front-end unit tests are heavily snapshot-based. Work that changes the DOM will result in a diff from the last accepted snapshot and cause related tests to fail. You can update the snapshots and review / accept the diff with `yarn test -u`.
+
+### End-to-end Tests with Cypress
+
+```
+cd app && yarn cypress
+```
+
+### Database Unit Tests with pgTAP
+
+```
+cd schema && make unit
+```
+
+## 9. Tidy Your Work
+
+[pre-commit](https://pre-commit.com/) runs a variety of formatting and lint checks configured in [`.pre-commit-config.yaml`](https://github.com/bcgov/cas-ciip-portal/blob/develop/.pre-commit-config.yaml) which are required for a pull request to pass CircleCI.
+
+`pre-commit install` will [configure a pre-commit hook to run before every commit](https://pre-commit.com/#usage); alternatively, you can run it manually with:
+
+```
+pre-commit run --all-files
+```
+
+If you are impatient and your work is isolated to Javascript, it may be faster to run only the related formatter (`xo`), but it may not catch everything (such as the end-of-file fixer and trailing whitespace):
```

Comment:
TODO (I may do this today of next time I do tech debt work): XO is too slow, switch back to plain eslint